packages/jazz-browser/src/auth/AnonymousAuth.ts

```diff
@@ -1,43 +1,37 @@
-import { AgentSecret } from "cojson";
+import { AgentSecret, CryptoProvider } from "cojson";
 import { Account, AuthMethod, AuthResult, ID } from "jazz-tools";
-
-type StorageData = {
-  accountID: ID<Account>;
-  accountSecret: AgentSecret;
-};
-
-const STORAGE_KEY = "jazz-logged-in-secret";
+import { AuthSecretStorage } from "./AuthSecretStorage.js";
 
 /**
- * `BrowserOnboardingAuth` provides a `JazzAuth` object for demo authentication.
+ * `BrowserAnonymousAuth` provides a `JazzAuth` object for demo authentication.
  *
  * Demo authentication is useful for quickly testing your app, as it allows you to create new accounts and log in as existing ones. The authentication persists across page reloads, with the credentials stored in `localStorage`.
  *
  * ```
- * import { BrowserOnboardingAuth } from "jazz-browser";
+ * import { BrowserAnonymousAuth } from "jazz-browser";
  *
- * const auth = new BrowserOnboardingAuth(driver);
+ * const auth = new BrowserAnonymousAuth(driver);
  * ```
  *
  * @category Auth Providers
  */
-export class BrowserOnboardingAuth implements AuthMethod {
+export class BrowserAnonymousAuth implements AuthMethod {
   constructor(
     public defaultUserName: string,
-    public driver: BrowserOnboardingAuth.Driver,
+    public driver: BrowserAnonymousAuth.Driver,
   ) {}
 
   /**
    * @returns A `JazzAuth` object
    */
-  async start() {
-    const existingUser = localStorage[STORAGE_KEY];
+  async start(crypto: CryptoProvider) {
+    AuthSecretStorage.migrate();
```

Comment:
Added a `migrate` call to move the auth systems to use the same credentials format